wp-content/themes/chrisdinh-photography/resources/scripts/components/handleGalleryCollectionFilterClick.js

```diff
@@ -0,0 +1,22 @@
+const handleGalleryCollectionFilterClick = () => {
+  const galleryCollections = document.querySelectorAll('button.gallery-collections__collection');
+
+  if (galleryCollections.length < 1) return;
+
+  galleryCollections.forEach(filter => {
```

Comment:
Consider event delegation: attaching a single event listener to the container of all buttons, galleryCollections, rather than attaching separate event listeners to each button.

Reply:
This file may be stale because we're using nanoGallery to handle filtering - i'll remove the file and leave a comment if that's the case.

But to your comment, i'm curious how that would work? How would you determine which filter should be the active filter if the event listener is added to the container instead of the individual filter @jonathanmdinh

Reply:
The handling of the buttons is about the same; the main difference is in how event listeners are attached.

With event delegation, you’d attach a single event listener to the container and use .closest() to listen for button clicks within it. However, you'd also have to handle cases where the user clicks within the container with the listener, but not on a button:

Example:
```
galleryCollectionsContainer.addEventListener('click', (e) => {
  const filter = e.target.closest('button.gallery-collections__collection');

  if (!filter) return; // Ignore clicks outside buttons

 // Continue with same handling.
  e.preventDefault();

  // Find buttons that have active class and remove it.
  document.querySelectorAll('.gallery-collections__collection--active').forEach(activeFilter => {
    activeFilter.classList.remove('gallery-collections__collection--active', 'border-b');
  });

  // Add active class to the clicked button, filter
  filter.classList.add('gallery-collections__collection--active');
});
```

This approach reduces the number of event listeners from n (one per button) to a single listener on the container. This allows for more scalability, especially if the number of buttons is dynamic without a cap. Theoretically, I think this would work.

Reply:
Ah that makes sense. I've deleted the file since we're using nanoGallery to handle filtering but if we do end up going a custom route in the future I'll be sure to use event delegation instead @jonathanmdinh